src/MongoDB.Bson/Serialization/BsonClassMap.cs

```diff
@@ -426,6 +416,55 @@ public static void RegisterClassMap(BsonClassMap classMap)
             }
         }
 
+        // private static methods
+        private static Type GetFormatterServicesType()
+        {
+#if NET452
+            return typeof(FormatterServices);
+#else
+            // TODO: once we depend on newer versions of .NET Standard we should be able to do this without reflection
+
+            try
+            {
+                // new approach which works on .NET Core 3.0
+                var formattersAssembly = Assembly.Load(new AssemblyName("System.Runtime.Serialization.Formatters"));
+                var formatterServicesType = formattersAssembly.GetType("System.Runtime.Serialization.FormatterServices");
+                if (formatterServicesType != null)
+                {
+                    return formatterServicesType;
+                }
+            }
+            catch
+            {
+                // ignore exceptions and continue to fallback code
+            }
+
+            // fallback to previous approach (which worked on older versions of .NET Core)
+            var mscorlibAssembly = typeof(string).GetTypeInfo().Assembly;
+            return mscorlibAssembly.GetType("System.Runtime.Serialization.FormatterServices");
+#endif
+        }
+
+        private static MethodInfo GetGetUninitializedObjectMethodInfo()
+        {
+            // don't let exceptions leak out of this method because it's called from the type initializer
+            try
+            {
+                var formatterServicesType = GetFormatterServicesType();
+                if (formatterServicesType != null)
+                {
+                    var formatterServicesTypeInfo = formatterServicesType.GetTypeInfo();
+                    return formatterServicesTypeInfo.GetMethod("GetUninitializedObject", BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Static);
+                }
+            }
+            catch
+            {
+                // ignore exceptions
+            }
+
+            return null;
```

Comment:
Are there already tests for a null return in consumers of this method?

Reply:
Yes. The return value of this method is stored in the static field __getUninitializedObjectMethodInfo. See line 38 above.

The only place __getUninitializedObjectMethodInfo is used checks to see if it is null:

https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Bson/Serialization/BsonClassMap.cs#L1252